### PR TITLE
Subgraph builders refactor

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Subgraph builders refactor

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/common/transformations/tests/common_optimizations/dimension_tracking.cpp [hunk 1]
```diff
@@ -19,6 +20,7 @@
 #include "transformations/common_optimizations/divide_fusion.hpp"
```
Comment (vurusovs) [inline src/common/transformations/tests/common_optimizations/dimension_tracking.cpp:20]: please, remove `subgraph_builders` include

File: src/plugins/auto_batch/tests/functional/behavior/plugin/auto_batching_tests.hpp [hunk 2]
```diff
@@ -12,6 +12,9 @@
 #include "common_test_utils/test_common.hpp"
 #include "functional_test_utils/blob_utils.hpp"
 #include "ov_models/subgraph_builders.hpp"
```
Comment (vurusovs) [inline src/plugins/auto_batch/tests/functional/behavior/plugin/auto_batching_tests.hpp:14]: please, remove `subgraph_builders`

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/bindings/c/tests/test_model_repo.cpp [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 #include <common_test_utils/file_utils.hpp>
 
+#include "common_test_utils/subgraph_builders/conv_pool_relu_no_reshapes.hpp"
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/bindings/c/tests/test_model_repo.cpp:9]: It produce too much compilation errors. I'd prefer to do in as separate commit

File: src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp [hunk 4]
```diff
@@ -32,7 +32,7 @@ std::shared_ptr<ov::Node> make_convolution(const ov::Output<Node>& in,
     if (!filter_weights.empty()) {
         filter_weights_node = std::make_shared<ov::op::v0::Constant>(type, filter_weights_shape, filter_weights);
     } else {
-        auto tensor = create_and_fill_tensor(type, filter_weights_shape);
+        auto tensor = create_and_fill_tensor(type, filter_weights_shape, 9, 1);
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp:35]: Why the change is required?

File: src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp [hunk 5]
```diff
@@ -10,6 +10,7 @@
 #include "common_test_utils/file_utils.hpp"
 #include "openvino/util/file_util.hpp"
 #include "functional_test_utils/summary/api_summary.hpp"
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp:12]: @sbalandi Could you please review the file and get rid OV1.0 based test classes?

File: src/tests/functional/plugin/shared/include/behavior/infer_request/set_blob_by_type.hpp [hunk 6]
```diff
@@ -42,8 +43,7 @@ class InferRequestSetBlobByType : public testing::WithParamInterface<InferReques
         // Skip test according to plugin specific disabledTestPatterns() (if any)
         SKIP_IF_CURRENT_TEST_IS_DISABLED()
         APIBaseTest::SetUp();
-        std::shared_ptr<ngraph::Function> function = ngraph::builder::subgraph::makeConvPoolRelu(
-                {4, 3, 6, 8}, ngraph::element::Type_t::u8);
+        std::shared_ptr<ngraph::Function> function = ov::test::utils::make_conv_pool_relu({4, 3, 6, 8}, ov::element::u8);
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/infer_request/set_blob_by_type.hpp:46]: `ngraph::Function` -> `ov::Model`

File: src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp (hunk 4 above)
Comment (iefode) [reply src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp:35]: to change default ranks in the case?

File: src/tests/functional/plugin/shared/include/behavior/infer_request/set_blob_by_type.hpp (hunk 6 above)
Comment (olpipi) [reply src/tests/functional/plugin/shared/include/behavior/infer_request/set_blob_by_type.hpp:46]: In scope of this PR I change only subgraph_builder related code

File: src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp:35]: @iefode Sorry, I didn't get what you mean

File: src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/activation.hpp [hunk 7]
```diff
@@ -1,6 +1,7 @@
 // Copyright (C) 2018-2023 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+#pragma once
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/activation.hpp:4]: To be fixed in next PR

File: src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp (hunk 4 above)
Comment (iefode) [reply src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp:35]: I totally agree with @vurusovs  and do not understand why you have changed the ranges.
In case we really need to update tensor filling, please update `generate_inputs` for Convolution

**Review by iefode — APPROVED** [review]
In general LGTM. Please fix @vurusovs comments and convolution case discussion

**Review by vurusovs — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/node_builders/convolution.cpp:35]: @iefode this is not about model parameters and inputs. This is about constants creation. We cannot operate with constants in generate_inputs. 
Indeed I made a mistake when I refactored these builders. And This change just fixes them.

If you want, we can have a quick call to clarify it.
